sp/Pedestrian.py

```diff
@@ -329,29 +329,38 @@ def tick(self, tick_count, delta_time, sim_time):
 
             # Calculate collision details if vehicle exists
             if collision_vehicle:
-                if self.get_collision_pt(vehicle_pos, vehicle_vel, self.path) is not None:
+                if self.collision_point is not None:
+                    collision_pt = [self.collision_point.x, self.collision_point.y]
+                    collision_segment_prev_node, collision_segment_next_node = self.get_curr_and_prev_path_nodes(self.path)
+
+                    # Euclidean distance between vehicle and collision point
+                    collision_vehicle_dist_to_collision = np.sqrt(np.sum((collision_pt - vehicle_pos) ** 2))
+                    time_to_collision = collision_vehicle_dist_to_collision / collision_vehicle.state.s_vel
+                    
+                elif self.get_collision_pt(vehicle_pos, vehicle_vel, self.path) is not None:
                     collision_pt, collision_segment_prev_node, collision_segment_next_node = self.get_collision_pt(vehicle_pos, vehicle_vel, self.path)
 
                     # Euclidean distance between vehicle and collision point
                     collision_vehicle_dist_to_collision = np.sqrt(np.sum((collision_pt - vehicle_pos) ** 2))
-                    if collision_vehicle.state.s_vel == 0:
-                        time_to_collision = float('inf')
-                    else:
-                        time_to_collision = collision_vehicle_dist_to_collision / collision_vehicle.state.s_vel
+                    time_to_collision = collision_vehicle_dist_to_collision / collision_vehicle.state.s_vel
+                
                 else:
                     collision_pt = None
```

Comment:
This is the default case (all values = None) that can be moved to line 319. Then these values can be overridden for collision vehicle and point. Also, remove the else branch 352 that repeats this default case.